README.md

```diff
@@ -6,6 +6,38 @@
 
 Needle is a dependency injection (DI) system for Swift. Unlike other DI frameworks, such as [Cleanse](https://github.com/square/Cleanse), [Swinject](https://github.com/Swinject/Swinject), Needle encourages **hierarchical DI structure and utilizes code generation to ensure compile-time safety**. This allows us to develop our apps and make changes with confidence. **If it compiles, it works.** In this aspect, Needle is more similar to [Android Dagger](https://google.github.io/dagger/).
 
+## The gist
+
+Using Needle to write DI code for your application is easy and compile-time safe. Each dependency scope is defined by a `Component`. And its dependencies are encapsualted in a Swift `protocol`. The two are linked together using Swift generics.
+
+```swift
+/// This protocol encapsulates the dependencies acquired from ancestor scopes.
+protocol MyDependency: Dependency {
+    /// These are objects obtained from ancestor scopes, not newly introduced at this scope.
+    var chocolate: Food { get }
+    var milk: Food { get }
+}
+
+/// This class defines a new dependency scope that can acquire dependencies from ancestor scopes
+/// via its dependency protocol, provide new objects on the DI graph by declaring properties,
+/// and instantiate child scopes.
+class MyComponent: Component<MyDependency> {
+
+    /// A new obejct, hotChocolate, is added to the dependency graph. Child scope(s) can then
+    /// acquire this via their dependency protocol(s).
+    var hotChocolate: Drink {
```

Comment:
nit: should be add `public` here to show that it's meant for use outside of this file ?

Reply:
It depends on how the project is structured really. For Uber internally, due to our module structure everything needs to be `public`. But I don't think that's a required case for most apps.